GliaWidgets/ViewController/EngagementViewController.swift

```diff
@@ -42,24 +42,24 @@ class EngagementViewController: ViewController, AlertPresenter {
         view.header.backButton.tap = { [weak self] in self?.viewModel.event(.backTapped) }
         view.header.closeButton.tap = { [weak self] in self?.viewModel.event(.closeTapped) }
 
-        viewModel.engagementAction = { action in
+        viewModel.engagementAction = { [weak self] action in
             switch action {
             case .confirm(let conf, let accessibilityIdentifier, confirmed: let confirmed):
-                self.presentConfirmation(
+                self?.presentConfirmation(
```

Comment:
Maybe here it would be worth it to do a `guard let self = self else { return }` instead to avoid having all the optional selfs? I don't know if that is more preferable.